components/builder-http-gateway/src/http/helpers.rs

```diff
@@ -429,6 +431,52 @@ pub fn get_optional_session_id(req: &mut Request) -> Option<u64> {
     }
 }
 
+pub fn generate_origin_encryption_keys(
```

Comment:
Minor - the function might read better if it was split into two  smaller ones vs. essentially repeating every line twice?  Not a blocker tho.